File: leetcode/617.合并二叉树/617.合并二叉树.py
from util.leetcode_type import TreeNode


class Solution:
    def mergeTrees(self, root1: TreeNode, root2: TreeNode) -> TreeNode:
        if root1 is None:
            return root2
        if root2 is None:
            return root1
        cur = TreeNode(root1.val + root2.val)
        cur.left = self.mergeTrees(root1.left, root2.left)
        cur.right = self.mergeTrees(root1.right, root2.right)
        return cur

File: leetcode/617.合并二叉树/617.合并二叉树.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func mergeTrees(root1 *TreeNode, root2 *TreeNode) *TreeNode {
	if root1 == nil {
		return root2
	}
	if root2 == nil {
		return root1
	}
	t3 := &TreeNode{root1.Val + root2.Val, nil, nil}
	t3.Left = mergeTrees(root1.Left, root2.Left)
	t3.Right = mergeTrees(root1.Right, root2.Right)
	return t3
}
